### DISCUSSION TITLE
New 'drivers/mss' subdirectory isn't reflected in associated module includes

### DISCUSSION THREAD

**Discussion by DrEamonnG** [polarfire-soc/polarfire-soc-documentation#57] [Bare metal embedded software] [answered]
I notice that some of the mss drivers have now been placed in an 'mss' subdirectory instead of being placed directly in the 'drivers' directory. For example, 'drivers/mss_ethernet_mac' is now 'drivers/mss/mss_ethernet_mac'. That in itself isn't a problem and I can see why that has been done. However, much of the source code, such as 'mss_ethernet_mac.c',  still have includes that looks like
"#include "drivers/mss_ethernet_mac/mss_ethernet_registers.h"
This causes problems when compiling the code.
What is the recommended solution for this? Go through and fix up all the includes with the correct path or remove the 'drivers/...' prefix and setup the compiler so that the path is sorted out at that level.

**Comment by nitindeshpande** [comment, marked as answer]
Hi, 
Thanks for pointing this out.
For now, you could choose any of the two options that you suggested which is easier for you. 
We have an updated version of the ethernet mac driver addressing this issue (and some other updates) which will be available shortly on the main branch.
